src/txn/txn.c

```diff
@@ -1024,8 +1024,11 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 
 		__wt_txn_op_free(session, op);
 	}
-	WT_ASSERT(session, skip_update_assert ||
-	    resolved_update_count == visited_update_count);
+	WT_ERR_ASSERT(session, skip_update_assert ||
+	    resolved_update_count == visited_update_count, EINVAL,
+	    "Number of resolved prepared updates: %" PRId64 " does not match"
```

Comment:
@sueloverso, I've updated the message to include counts. I'm concerned the message is a bit cryptic thought but I'm not really sure how I could make it more meaningful. What do you think?

Reply:
I think if/when the message triggers it will be helpful to the debuggers. It is no more cryptic than the same message but without the values. The first thing someone debugging this assertion failure is going to want to see is what those two values are. Thanks for adding it.